### ISSUE TITLE
LCD Timer resets to 00:00 after 24 Hour Print

### ISSUE THREAD

**Issue by psavva** [MarlinFirmware/Marlin#4565]
The Timer on the LCD Resets back to 00:00 after 24 Hour Print.
ie: 25 Hours will become 01:00 and not 1day 01H 00 MI or alternatively 25:00  (HH:MI) or similar

**Comment by thinkyhead** [comment]
Are you referring to the clock on the **Info Screen**? We can get up to 99:59 with…

``` diff
  void toDigital(char *buffer) const {
-   int h = this->hour() % 24,
+   int h = this->hour(),
        m = this->minute() % 60;

    sprintf_P(buffer, PSTR("%02i:%02i"), h, m);
  }
```

**Comment by psavva** [comment]
I guess this must be from dogm_lcd_implementation.h
 Line ~375

```
    uint16_t time = print_job_timer.duration() / 60;
    if (time != 0) {
      lcd_print(itostr2(time/60));
      lcd_print(':');
      lcd_print(itostr2(time%60));
```

Have a look at this image:
https://plus.google.com/u/0/photos/112582746481845934091/album/6316677531307235121/6316677534972389490?authkey=CPXFieGzkuegZg&sqid=117814474100552114108&ssid=243dc826-ec08-43e8-9d60-3103ba5b6e21

You'll notice that at the top of the progress bar, you'll notice that it says 20:02 - Implies 20 Hours and 2 Mins.
However, it actually printed for 1 Day 20 Hours and 2 Mins...

So the 20:02 should have been 44:02 to show total of 44 Hours, and 22 Mins.

If I can get a mentor on how to debug Marlin, and how to trace the code, I'd be happy to investigate the issue, and find the solution :)

Every attempt I've made to fix something has been somewhat of a failure, as I cannot see how to step into code, how to go about line by line debugging...  
If someone could mentor me, that would be great!

**Comment by thinkyhead** [comment]
I see. There's more space on the _graphical_ display, so a days count could fit there. On the character-based display we only have space for hh:mm. Anyway, I already patched it to show up to 99:59 with #4572, as shown above.

This is in `RCBugFix` of course. We aren't patching older code. For an older version of Marlin, take the number of seconds, divide by 86400 to get the number of days. To get the number of hours, without wrapping around, just remove the modulus `% 24` from wherever it's printing the hour value.

**Comment by boelle** [comment]
are we keeping this to add days also?

**Comment by jbrazio** [comment]
@boelle Fixed by #4572. (Scroll the page to see above.)

@psavva Please reopen this ticket if the problem still persists.
